src/graphql/subscription/map_async_iterator.py

```diff
@@ -43,9 +43,18 @@ async def __anext__(self) -> Any:
             aclose = ensure_future(self._close_event.wait())
             anext = ensure_future(self.iterator.__anext__())
 
-            pending: Set[Future] = (
-                await wait([aclose, anext], return_when=FIRST_COMPLETED)
-            )[1]
+            # Suppress the StopAsyncIteration exception warning when the iterator is cancelled.
+            anext.add_done_callback(lambda *args: anext.exception())
+            try:
+                pending: Set[Future] = (
+                    await wait([aclose, anext], return_when=FIRST_COMPLETED)
+                )[1]
+            except CancelledError:
+                # The iterator is cancelled
+                aclose.cancel()
+                anext.cancel()
+                raise StopAsyncIteration
```

Comment:
You could use "raise from" here.

If possible, add a test that executes these lines so that we have full coverage. Otherwise, add "# pragma: nocover" for the time being, Maybe I will add a test then later.

Reply:
I will try to add the test and fix the code style issue and let you know. Thanks

Reply:
@Cito Can you suggest how to run the coverage test? I ran `tox testenv` on a windows Python 3.8 and I got 
```
manifest run-test: commands[0] | check-manifest -v
listing source files under version control: 316 files and directories
building an sdist
['c:\\users\\wuyua\\projects\\pycharmprojects\\graphql-core\\.tox\\manifest\\scripts\\python.exe', '-m', 'build', '--sdist', '.', '--outdir', 'C:\\Users\\wuyua\\AppData\\Local\\Temp\\check-manifest-mv8ko07e-sdist'] failed (status 1)
:
Unable to copy 'c:\\users\\wuyua\\projects\\pycharmprojects\\graphql-core\\.tox\\manifest\\scripts\\venvlauncher.exe'
Unable to copy 'c:\\users\\wuyua\\projects\\pycharmprojects\\graphql-core\\.tox\\manifest\\scripts\\venvlauncher.exe'
Unable to copy 'c:\\users\\wuyua\\projects\\pycharmprojects\\graphql-core\\.tox\\manifest\\scripts\\venvwlauncher.exe'
Traceback (most recent call last):
  File "C:\Users\wuyua\anaconda3\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\wuyua\anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "c:\users\wuyua\projects\pycharmprojects\graphql-core\.tox\manifest\lib\site-packages\build\__main__.py", line 214, in <module>
    main(sys.argv[1:], 'python -m build')
  File "c:\users\wuyua\projects\pycharmprojects\graphql-core\.tox\manifest\lib\site-packages\build\__main__.py", line 206, in main
    build_package(args.srcdir, outdir, distributions, config_settings, not args.no_isolation, args.skip_dependencies)
  File "c:\users\wuyua\projects\pycharmprojects\graphql-core\.tox\manifest\lib\site-packages\build\__main__.py", line 94, in build_package
    _build_in_isolated_env(builder, outdir, distributions, config_settings)
  File "c:\users\wuyua\projects\pycharmprojects\graphql-core\.tox\manifest\lib\site-packages\build\__main__.py", line 52, in _build_in_isolated_env
    with IsolatedEnvBuilder() as env:
  File "c:\users\wuyua\projects\pycharmprojects\graphql-core\.tox\manifest\lib\site-packages\build\env.py", line 91, in __enter__
    executable, scripts_dir = _create_isolated_env_venv(self._path)
  File "c:\users\wuyua\projects\pycharmprojects\graphql-core\.tox\manifest\lib\site-packages\build\env.py", line 199, in _create_isolated_env_venv
    venv.EnvBuilder(with_pip=True).create(path)
  File "C:\Users\wuyua\anaconda3\lib\venv\__init__.py", line 68, in create
    self._setup_pip(context)
  File "C:\Users\wuyua\anaconda3\lib\venv\__init__.py", line 294, in _setup_pip
    subprocess.check_output(cmd, stderr=subprocess.STDOUT)
  File "C:\Users\wuyua\anaconda3\lib\subprocess.py", line 411, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "C:\Users\wuyua\anaconda3\lib\subprocess.py", line 489, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\Users\wuyua\anaconda3\lib\subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\wuyua\anaconda3\lib\subprocess.py", line 1307, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

ERROR: InvocationError for command 'C:\Users\wuyua\Projects\PycharmProjects\graphql-core\.tox\manifest\Scripts\check-manifest.EXE' -v (exited with code 2)
```

Reply:
E.g. `tox -e py39` should work, or just `tox` to run all the tests.